BL-4819 Device css for margins, pg numb

gmartin7 opened the pull request:
* install DeviceBottomMargin variable and mixin
* add device.less to import into basePage
* center page number on Device layouts
* leave some space at the bottom of Device layouts
   for page number, if needed

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bloombooks/bloomdesktop/1779)
<!-- Reviewable:end -->

andrew-polk commented:
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/BloomBrowserUI/bookLayout/basePage.less, line 181 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779#-Kqik5Z91IJszFtULNik:-Kqik5Z91IJszFtULNil:b1069wn) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/815aceb25db7065f0535b7a2908a862a103914d2/src/BloomBrowserUI/bookLayout/basePage.less#L181)):*
> ```Less
> */
> .bloom-page {
>     // See device.less for all other Device and PictureStory layout css
> ```

Any reason not move these there, too?

---

*[src/BloomBrowserUI/bookLayout/device.less, line 5 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779#-KqijatfeRSE2zE4wUcl:-KqijatgW6YhPSRlyoFm:bygluuf) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/815aceb25db7065f0535b7a2908a862a103914d2/src/BloomBrowserUI/bookLayout/device.less#L5)):*
> SetMarginBoxDeviceHeight

If this only applies when we have a page number, I would rather name it something which reflects that.

---

*[src/BloomBrowserUI/bookLayout/device.less, line 6 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779#-KqiiR4UtYnYu1SJqZpn:-KqiiR4UtYnYu1SJqZpo:b4pl23d) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/815aceb25db7065f0535b7a2908a862a103914d2/src/BloomBrowserUI/bookLayout/device.less#L6)):*
> !important

is this required? If so, do we know why?
Same question for the ones used to center the page number.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779)*
<!-- Sent from Reviewable.io -->

gmartin7 commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/BloomBrowserUI/bookLayout/basePage.less, line 181 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779#-Kqik5Z91IJszFtULNik:-KqilKiBC5I_3rrjY4Qv:bin1o7e) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/815aceb25db7065f0535b7a2908a862a103914d2/src/BloomBrowserUI/bookLayout/basePage.less#L181)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Any reason not move these there, too?
</blockquote></details>

Mostly I wanted to avoid losing track of them in this file, but I suppose the comment can be updated to serve that purpose.

---

*[src/BloomBrowserUI/bookLayout/device.less, line 5 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779#-KqijatfeRSE2zE4wUcl:-KqinNO4QvYKhc7OGM6t:bj9q0n3) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/815aceb25db7065f0535b7a2908a862a103914d2/src/BloomBrowserUI/bookLayout/device.less#L5)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> SetMarginBoxDeviceHeight

If this only applies when we have a page number, I would rather name it something which reflects that.
</blockquote></details>

You're right. Changed to "LeaveMarginForPageNumber".

---

*[src/BloomBrowserUI/bookLayout/device.less, line 6 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779#-KqiiR4UtYnYu1SJqZpn:-Kqip0bedO3wg4AWOqEJ:b-5swbqo) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/815aceb25db7065f0535b7a2908a862a103914d2/src/BloomBrowserUI/bookLayout/device.less#L6)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> !important

is this required? If so, do we know why?
Same question for the ones used to center the page number.
</blockquote></details>

This one is removed. I left the !important on the left: rule for centering because basePage's default rule has the same precedence and is later in the file (due to importing at the top?). The one for right is okay without it, since I'm adding a class to the existing rule. Also removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779)*
<!-- Sent from Reviewable.io -->

andrew-polk commented:
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomBrowserUI/bookLayout/device.less, line 72 at r2](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779#-KqiryvuSnPYizUaCu0V:-KqiryvuSnPYizUaCu0W:b-5piox3) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c2de82f3f611f50eb659d970df9fc5e45e5d0bef/src/BloomBrowserUI/bookLayout/device.less#L72)):*
> ```Less
>     &.numberedPage{
>         &::after{
>             // the following rule has the same precedence as the default rule for this case
> ```

I could be wrong, but I would suspect the other one gets precedence because it is specifying side-left.
Otherwise, I would expect this one to win because it specifying .Device16x9Landscape.
Would be worth a quick test. And might be worth including side-left here, anyway. AFAIK, all pages should either be side-left or side-right now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779)*
<!-- Sent from Reviewable.io -->

andrew-polk commented:
*[src/BloomBrowserUI/bookLayout/device.less, line 72 at r2](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779#-KqiryvuSnPYizUaCu0V:-KqiuieX4nU6Fwqv_RUu:b-na7qgg) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c2de82f3f611f50eb659d970df9fc5e45e5d0bef/src/BloomBrowserUI/bookLayout/device.less#L72)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

I could be wrong, but I would suspect the other one gets precedence because it is specifying side-left.
Otherwise, I would expect this one to win because it specifying .Device16x9Landscape.
Would be worth a quick test. And might be worth including side-left here, anyway. AFAIK, all pages should either be side-left or side-right now.
</blockquote></details>

On second thought, can we just use the same rule for side-left and side-right since we are trying to center anyway?
I mean, you might still have to specify to get it to override the inherited rule, but you may be able to do so with a comma.
Or am I missing why side-left and side-right are centered differently?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779)*
<!-- Sent from Reviewable.io -->

gmartin7 commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomBrowserUI/bookLayout/device.less, line 72 at r2](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779#-KqiryvuSnPYizUaCu0V:-Kqiuz7Qtj9LyQXtGmGz:b-cylipn) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c2de82f3f611f50eb659d970df9fc5e45e5d0bef/src/BloomBrowserUI/bookLayout/device.less#L72)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

On second thought, can we just use the same rule for side-left and side-right since we are trying to center anyway?
I mean, you might still have to specify to get it to override the inherited rule, but you may be able to do so with a comma.
Or am I missing why side-left and side-right are centered differently?
</blockquote></details>

Yes, that's better after adding side-left. The reason it was not there was to make the bottom rule apply to both, but I've just factored that out. They have to be centered differently because each is overriding a rule that works on (respectively) left and right.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779)*
<!-- Sent from Reviewable.io -->

andrew-polk commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1779#-:-Kqivy23hpapr1OeenGY:bnfp4nl)*
<!-- Sent from Reviewable.io -->
